src/DateTimeField.js

```diff
@@ -100,13 +105,41 @@ export default class DateTimeField extends Component {
 
   }
 
+  checkIsValid = (value) => {
+    return moment(value, this.state.inputFormat, true).isValid() || value === this.props.defaultText || value === '';
+  }
+
+  setIsValid = (isValid) => {
+    return this.setState({
+      isValid: isValid
+    })
+  }
+
   getValue = () => {
     return moment(this.state.inputValue, this.props.inputFormat, true).format(this.props.format);
   }
 
+  setSelectedMonth = (e) => {
+    const { target } = e;
+    if (target.className && !target.className.match(/disabled/g)) {
+      return this.setState({
+        selectedDate: this.state.viewDate.clone().month(e.target.innerHTML).date(this.state.selectedDate.date()).hour(this.state.selectedDate.hours()).minute(this.state.selectedDate.minutes())
+      }, function() {
+        this.closePicker();
+        this.props.onChange(this.state.selectedDate.format(this.props.format));
+        return this.setState({
+          inputValue: this.state.selectedDate.format(this.state.inputFormat)
+        });
+      });
+    }
+  }
+
   setSelectedDate = (e) => {
     const { target } = e;
+    const value = target == null ? event : event.target.value;
```

Comment:
@maggie-mu Should this say:

`const value = target == null ? e : target.value;`